language-design.md

```diff
@@ -17,6 +17,56 @@ For each quote, describe what it was about the quote that led you pick it.
+with other tools (like classes), doesn't necessarily mean we shouldn't give them both tools, 
+especially if it reduces "boilerplate..., which is annoying and error-prone" [Bloch, 2006]. The set 
+of tools must be carefully curated to avoid warts, both of the missing and tacked-on varieties.
+
+"That’s not Greek, it’s Klingon. The Perl version uses fewer characters, which many geeks would no 
+doubt see as more efficient or precise; but Stefik says that the Quorum version accomplishes 
+exactly the same commands" [Pavlus, 2012].
+
+The author here is for some reason conflating "fewer characters" with "more efficient or precise." 
+What the author lauds as "more readable code" is often just code that has its syntactic sugar 
+removed. The Quorum for-loop is really just a while loop with an unrelated variable being updated 
+inside it; this might be easier to grok for a brand-new programmer, but it slows down the 
+comprehension for anyone above that level. The point of syntactic sugar, like for...in loops in 
+JavaScript, is to simplify the wording of something every programmer already knows how to do, to 
+allow it to be written and read much faster. As someone who originally started programming with 
+MIT's Scratch then moved to other languages with C-like syntax, it feels weirdly infantilizing to 
+demand that all languages be absolutely readable to any person. This seems like demanding that the 
+entire legal code be rewritten using only the thousand most common English words, simpleWikipedia 
+style. I'm all for making CS an easier field to enter, and making syntax less arcane; however, I 
+feel like tools like Scratch and Jupyter notebooks already fill that niche; there's no need to 
+completely replace code with lines of prose.
+
+"We should remember that 'knowing' a language is a poor proxy for being able to think rigorously. 
+It is one thing to have learned the syntax of a language and another to have grasped the underlying 
+paradigms" [Yang and Rabkin, 2015].
+
+The way the CS job market is structured really incentivizes this sort of language collection. 
+People in tech, especially recent graduates, try to pad out their resume with as many languages as 
+possible: it doesn't matter if you've only used C# once, it's going on your resume to make the 
+"Technologies" section bigger and to give you an edge over the other applicants. I'm hoping this 
+is, in some ways, partially starting to shift. I've seen quite a few jobs demanding "strong 
+reasoning skills" and "experience in object-oriented programming", rather than a laundry list of 
+C-based languages. One of the things I think Harvey Mudd's CS department does best is emphasizing 
+rigorous thinking about problems in abstract ways, instead of just in terms of how languages can 
+solve them. CS5, CS81, and CS140 especially have improved the way I think about problems in 
+general, even those outside of the CS field.
+
```

Comment:
I agree! This is one of the biggest shifts I saw from high school and college cs, was Mudd had a completely different emphasis from learning a bunch of languages.